modules/seqexec/server/src/main/scala/seqexec/server/flamingos2/Flamingos2ControllerEpics.scala

```diff
@@ -148,8 +134,22 @@ object Flamingos2ControllerEpics extends Flamingos2Controller {
     _ <- Flamingos2Epics.instance.endObserveCmd.post
   } yield ()
 
-  override def observeProgress(total: Time): fs2.Stream[IO, Progress] =
-    ProgressUtil.countdown[IO](total, Seconds(0))
+  override def observeProgress(total: Time): fs2.Stream[IO, Progress] = {
+    val s = ProgressUtil.fromStateTOption[IO, Time](_ => StateT[IO, Time, Option[Progress]] { st =>
+      IO {
+        val m = if (total >= st) total else st
+        val p = for {
+          obst <- Flamingos2Epics.instance.observeState
+          if obst.isBusy
+          rem <- Flamingos2Epics.instance.countdown.map(_.seconds)
+        } yield Progress(m, RemainingTime(rem))
+        (m, p)
+      }
+    })
+    s(total).dropWhile(_.remaining.self.value === 0.0) // drop leading zeros
+      .takeThrough(_.remaining.self.value > 0.0) // drop all tailing zeros but the first one
+
+  }
```

Comment:
FLAMINGOS-2 starts with a countdown value higher than the exposure time. I used a `State` to carry that value as the new `total`.